src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -190,37 +190,40 @@ private AssemblyNameParts Parse()
 
         private Version ParseVersion(string attributeValue)
         {
-            string[] parts = attributeValue.Split('.');
-            if (parts.Length > 4)
+            ReadOnlySpan<char> attributeValueSpan = attributeValue;
+            Span<Range> parts = stackalloc Range[5];
+            parts = parts.Slice(0, attributeValueSpan.Split(parts, '.'));
+            if (parts.Length is not (3 or 4) ||
+                attributeValue.Contains('\0')) // ushort.TryParse below will ignore trailing nulls, so they're flagged here
```

Comment:
`attributeValue.Contains('\0')` should not be needed. The nulls are filtered out earlier in `GetNextToken` method.